src/diggler/command.d

```diff
@@ -422,16 +424,127 @@ abstract class CommandSet(T) : ICommandSet
 
 	final void registerCommands(T cmdSet)
 	{
-		foreach(memberName; __traits(derivedMembers, T))
+		import diggler.std_backport.meta : staticSort;
+		import std.meta : Filter;
+		import std.algorithm : canFind, commonPrefix, joiner, splitter;
+		import std.range : take, walkLength;
+		import std.string : startsWith, endsWith;
+		import std.conv : to;
+
+		//The separator is strictly a string: any other type will halt compilation
+		static if(__traits(hasMember, T, "subCommandSeparator") && is(typeof(T.subCommandSeparator) == string))
+			enum cmdSep = T.subCommandSeparator;
+		else static if(__traits(hasMember, T, "subCommandSeparator") && !is(typeof(T.subCommandSeparator) == string))
+			static assert(false, "subCommandSeparator must be of type string, not " ~ typeof(T.subCommandSeparator).stringof);
+		else
+			enum cmdSep = "_";
+
+		//Determines whether a given symbol name is a valid command
+		template isCommand(string symbol){
+			static if(__traits(getProtection, __traits(getMember, T, symbol)) != "public")
+				enum isCommand = false;
+			else
+				enum isCommand =
+					symbol != "__ctor" &&
+					symbol != "__dtor" &&
+					__traits(compiles, __traits(getMember, T, symbol)) && // ahem...
```

Comment:
I don't think this can be false now that `getProtection` is fixed and handles this?